File: app/src/main/java/com/sakurafish/pockettoushituryou/view/activity/WebViewActivity.java
package com.sakurafish.pockettoushituryou.view.activity;

import android.content.Context;
import android.content.Intent;
import androidx.databinding.DataBindingUtil;
import android.os.Bundle;
import androidx.annotation.NonNull;

import com.sakurafish.pockettoushituryou.R;
import com.sakurafish.pockettoushituryou.databinding.ActivityWebviewBinding;
import com.sakurafish.pockettoushituryou.view.fragment.WebViewFragment;

public class WebViewActivity extends BaseActivity {

    private static final String TAG = WebViewActivity.class.getSimpleName();
    public static final String EXTRA_URL = "url";
    public static final String EXTRA_TITLE = "title";

    ActivityWebviewBinding binding;

    public static Intent createIntent(@NonNull Context context, @NonNull String url, @NonNull String title) {
        Intent intent = new Intent(context, WebViewActivity.class);
        intent.putExtra(EXTRA_URL, url);
        intent.putExtra(EXTRA_TITLE, title);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        final String url = getIntent().getStringExtra(EXTRA_URL);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_webview);
        initView();
        replaceFragment(WebViewFragment.newInstance(url), R.id.content_view);
    }

    private void initView() {
        setSupportActionBar(binding.toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);
        getSupportActionBar().setTitle(getIntent().getStringExtra(EXTRA_TITLE));
        binding.toolbar.setNavigationOnClickListener(view -> onBackPressed());
    }
}

File: app/src/main/java/com/sakurafish/pockettoushituryou/view/activity/WebViewActivity.kt
package com.sakurafish.pockettoushituryou.view.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import com.sakurafish.pockettoushituryou.R
import com.sakurafish.pockettoushituryou.databinding.ActivityWebviewBinding
import com.sakurafish.pockettoushituryou.view.fragment.WebViewFragment

class WebViewActivity : AppCompatActivity() {

    private lateinit var binding: ActivityWebviewBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_webview)
        initView()
    }

    private fun initView() {
        setSupportActionBar(binding.toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setDisplayShowHomeEnabled(true)
        supportActionBar!!.title = intent.getStringExtra(EXTRA_TITLE)
        binding.toolbar.setNavigationOnClickListener { onBackPressed() }

        val url = intent.getStringExtra(EXTRA_URL)
        val fragment: Fragment? = supportFragmentManager.findFragmentById(R.id.fragment_webview)
        if (fragment is WebViewFragment) {
            // TODO Replace Navigation safe args
            fragment.setUrl(url)
        }
    }

    companion object {
        const val EXTRA_URL = "url"
        const val EXTRA_TITLE = "title"

        fun createIntent(context: Context, url: String, title: String): Intent {
            val intent = Intent(context, WebViewActivity::class.java)
            intent.putExtra(EXTRA_URL, url)
            intent.putExtra(EXTRA_TITLE, title)
            return intent
        }
    }
}
